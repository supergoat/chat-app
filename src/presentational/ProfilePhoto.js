// @flow
import React, { Component } from 'react';
import {
  Text,
  View,
  Image
} from 'react-native';
import CAColors from "../common/CAColors";
import CAStyleSheet from "../common/CAStyleSheet";
import profilePhotoPlaceholder from '../assets/img/profile_photo_placeholder.png';

/* Constants
============================================================================= */
const PROFILE_PHOTO_HEIGHT = 50;
const PROFILE_PHOTO_WIDTH = PROFILE_PHOTO_HEIGHT;
const PROFILE_PHOTO_BORDER_RADIUS = PROFILE_PHOTO_HEIGHT / 2;

/** ============================================================================
<ProfilePhoto
  [profilePhoto]={string}
  name={string}
  [style]={object}
/>
--------------------------------------------------------------------------------
The profile photo of the user
@param { string } [profilePhoto] - The profile photo of the person
@param { string } name - The name of the person
@param { object } style - custom style
============================================================================= */
type Props = {
  profilePhoto?: string,
  name: string,
  style?: { [name: string]: any }
}
class ProfilePhoto extends Component<Props> {

  static __cards__: Function

  render() {
    const { profilePhoto, name, style } = this.props;
    return (
      <Image
        source={profilePhoto ? {uri: profilePhoto} : profilePhotoPlaceholder}
        style={[styles.profilePhoto, style]}
        alt={name}
      />
    )
  }
}

/* StyleSheet
============================================================================= */
const styles = CAStyleSheet.create({
  profilePhoto: {
    backgroundColor: CAColors.gallery,
    borderRadius: PROFILE_PHOTO_BORDER_RADIUS,
    height: PROFILE_PHOTO_HEIGHT,
    width: PROFILE_PHOTO_WIDTH,
  }
});

/* Playground Cards - the different states of ProfilePhoto
============================================================================= */
ProfilePhoto.__cards__ = define => {
  define("Without Profile Photo", () => {
    return (
      <ProfilePhoto
        name="James Gordon"
      />
    )
  });

  define("With Profile Photo", () => {
     return (
       <ProfilePhoto
         profilePhoto='https://placekitten.com/60/60'
         name="Alex Stone"
       />
     )
  });
};


/* Export
============================================================================= */
export default ProfilePhoto;
