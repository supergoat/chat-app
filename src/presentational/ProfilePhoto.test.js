// @flow
import React from 'react';
import ProfilePhoto from './ProfilePhoto';
import renderer from 'react-test-renderer';

describe('ProfilePhoto', () => {
  it('displays the profilePhoto when its provided', () => {
    const tree = renderer.create(
      <ProfilePhoto
        profilePhoto='https://placekitten.com/60/60'
        name="Alex Stone"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays a placeholder image when the profilePhoto is NOT provided', () => {
    const tree = renderer.create(
      <ProfilePhoto
        name="James Gordon"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
