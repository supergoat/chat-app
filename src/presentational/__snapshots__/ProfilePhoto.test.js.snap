// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePhoto displays a placeholder image when the profilePhoto is NOT provided 1`] = `
<Image
  alt="James Gordon"
  source={
    Object {
      "testUri": "../../../src/assets/img/profile_photo_placeholder.png",
    }
  }
  style={
    Array [
      Object {
        "backgroundColor": "#EEEEEE",
        "borderRadius": 25,
        "height": 50,
        "width": 50,
      },
      undefined,
    ]
  }
/>
`;

exports[`ProfilePhoto displays the profilePhoto when its provided 1`] = `
<Image
  alt="Alex Stone"
  source={
    Object {
      "uri": "https://placekitten.com/60/60",
    }
  }
  style={
    Array [
      Object {
        "backgroundColor": "#EEEEEE",
        "borderRadius": 25,
        "height": 50,
        "width": 50,
      },
      undefined,
    ]
  }
/>
`;
